libfwupdplugin/fu-self-test.c

```diff
@@ -695,28 +695,34 @@ fu_device_locker_func (void)
 }
 
 static gboolean
-_fail_open_cb (GObject *device, GError **error)
+_fail_open_cb (FuDevice *device, GError **error)
 {
+	fu_device_set_metadata_boolean (device, "isLocked", TRUE);
```

Comment:
Shouldn't this be heavier than a metadata item? I feel like it should be a private property with methods to call.

Reply:
`FU_DEVICE_INTERNAL_FLAG_LOCKED` seems sane, no?

Reply:
seems good

Reply:
> FU_DEVICE_INTERNAL_FLAG_LOCKED seems sane, no?

Turns out, the answer was indeed: **no**. Let me explain: The "is-locked" is a property of the locker, not of the device. It's quite valid to (and plugins do) call something like:

    locker1 = fu_device_locker_new_full (device, _open_cb, _close_cb, error)
    locker2 = fu_device_locker_new_full (device, _detach_cb, _attach_cb, error)

So if we set the flag on the device for the first locker, then it would be invalid for the second. I then pondered `fu_device_locker_is_open()` but all that really tells you is whether `fu_device_locker_close` has been manually called, as if the `->setup()` failed in the locker open the locker returns NULL. Notably, we also call `->close_func` on locker finalize, so we could have just moved up the `self->device_open = TRUE;` but in this case I think handling it locally is better as we get the `g_debug("ignoring close error on aborted open")` rather than the `g_warning()` on finalize.

Reply:
I did the pragmatic thing. New version below.